### ISSUE TITLE
SDL2 windows open on wrong output

### ISSUE THREAD

**Issue by HEX-01** [swaywm/sway#4245]
```
$ swaymsg -t get_version
sway version 1.0
```
```
$ pacman -Q sway
sway 1.1.1
```

sway.log and config
https://gist.github.com/HEX-01/cc17775df7c950f055548bd3c41dd51d

[gif-screencast](https://imgur.com/NS2N1KO)

My laptop 1366x768 is hooked up 1920x1080 monitor. I use 1080p monitor as primary monitor. When I try to launch sdl2 applications which are running as wayland native they all open on the laptop screen (0ad, xonotic-sdl). I can move them to 1080p workspaces but they won't resize properly (window box/decoration resizes but the contents stay the same size as can be seen in the gif recording). Mouse pointer also doesn't "target" properly as can be seen in the gif.

It seems like some applications select the primary display if there's such a thing in wayland.
I found a workaround that fixes this problem:
`exec sleep 1 && swaymsg output LVDS-1 disable && swaymsg output LVDS-1 enable`

EDIT: it doesn't fix it. It just makes it so SDL2 wayland windows always open on the external 1920x1080 monitor.

**Comment by mcoffin** [comment]
Have you tried setting `SDL_VIDEODRIVER=wayland` as per the [instructions on the ArchLinux wiki](https://wiki.archlinux.org/index.php/Wayland#SDL2)?

**Comment by HEX-01** [comment]
> Have you tried setting `SDL_VIDEODRIVER=wayland` as per the [instructions on the ArchLinux wiki](https://wiki.archlinux.org/index.php/Wayland#SDL2)?

Yes I have. I set it in /etc/environment and as can be seen in the gif xeyes doesn't track the mouse cursor inside 0ad window so it's not running in xwayland. `env` also reports `SDL_VIDEODRIVER=wayland`.
And weirdly  enough disabling and enabling laptop output after starting sway makes SDL2 windows always open on the external 1080p monitor.

**Comment by mcoffin** [comment]
So, I see you're launching the application with `dmenu`, yes? `dmenu` is an X11 application, which is probably what's going on. Try with some wayland launcher, like what I have set up (on top of my terminal, alacritty).

```
bindsym $mod+Shift+d exec alacritty --class 'launcher' --command bash -c 'compgen -c | sort -u fzf | xargs -r swaymsg -t command exec'
for_window [app_id="^launcher$"] floating enable, border none, resize set width 25 ppt height 100 ppt, move position 0 px 0 px
```
With that config, try launching with `$mod+Shift+d`.

**Comment by HEX-01** [comment]
> So, I see you're launching the application with `dmenu`, yes?
No I was launching it with bemenu with env variable `BEMENU_BACKEND=wayland`
Ant it's indeed running as wayland native.
